src/basic/terminal-util.c

```diff
@@ -1565,6 +1545,32 @@ int set_terminal_cursor_position(int fd, unsigned int row, unsigned int column)
         return 0;
 }
 
+int terminal_enable_line_wrapping_fd(int fd) {
+        int r, k;
+
+        assert(fd >= 0);
+
+        if (terminal_is_dumb())
```

Comment:
The switch from `terminal_is_dumb()` to `getenv_terminal_is_dumb()` somehow causes the terminal to be reset even with a `console=ttyO0,115200n8` kernel command line, which is unfortunate on embedded systems because it also erases all the messages before that came from the bootloader. Was this the intended behaviour? Otherwise I'd send a patch switching it back to plain `terminal_is_dumb()`.

Reply:
@rohieb Check out #34571

Reply:
Actually nvm that wouldn't work. Can you open a separate issue for this explaining your setup in more detail?